addon/components/frost-numeric-select.js

```diff
@@ -0,0 +1,183 @@
+         return false
+       }
+      return true
+    } else {
+      if (charCode == 45 && !this.get('value')) {//check if user can enter negative symbol
+        return true
+      } else if (charCode == 46 && decimalExist) {
+        return false
+      } else if (charCode != 46 && charCode > 31
+        && (charCode < 48 || charCode > 57))
+         return false
+
+      return true
+    }
+  },
+
+  //calculate how many digit a value has
+  _calculateDigit (value) {
+    return value.slice(value.indexOf('.')+1).length
+  },
+
+  //set calculated and properly formated value
+  _setCalculatedValue (value, numberOfDigit) {
+    if(this.$('input').val().indexOf('.') > -1) {
+      this.set('value', value.toFixed(numberOfDigit).toString())
+    } else {
+      this.set('value', value.toString())
+    }
+  },
+
+  // == Computed Properties ===================================================
+  /**
+   * Determine which scss class should be used
+   * @param {boolean} isArrowVisible - if the user allow displaying the up and down arrow buttons
+   * @returns {string} - a scss class name
+   */
+  @computed('isArrowVisible')
+  class (isArrowVisible) {
+    return isArrowVisible ? 'frost-numeric-select-button' : 'frost-numeric-select'
+  },
+
+  // == Lifecycle Hooks =======================================================
+  init () {
+    this._super(...arguments)
+    this.set('receivedHook', this.get('hook'))
+    if (this.get('isHookEmbedded')) {
+      this.set('hook', '')
+    }
+  },
+  // == Actions ===============================================================
+
+  actions: {
+    keyUp (value, event) {
+      if (isPresent(this.get('_eventProxy.keyUp'))) {
+        this._eventProxy.keyUp(event)
+      }
+    },
+
+    _onKeyPress (event) {
+      if (!this._isNumberKey(event)) {
+          this.set('showError', true)
+          event.preventDefault();
+      }
+      else {
+        this.set('showError', false)
+      }
+    },
+
+    _onInput (event) {
+      if (isPresent(this.get('_eventProxy.input'))) {
+        // Add id and value for legacy support
+        event.id = this.get('elementId')
+        event.value = event.target.value
+        this._eventProxy.input(event)
+      }
+    },
+    _onPlusClick (event) {
+      if (this.$('input').val()){
+        //calculate how many digit a value has
+        let numberOfDigit = this._calculateDigit(this.$('input').val())
+        var value = parseFloat(this.get('value'))+1
+
+        if (!this.get('isAllowUseDownButtonToNegative')) {
+          if(value >= 0)
+            this._setCalculatedValue(value,numberOfDigit)
+        } else {
+          this._setCalculatedValue(value,numberOfDigit)
+        }
+      }
+    },
+    _onMinusClick (event) {
+      if (this.$('input').val()){
+        let numberOfDigit = this._calculateDigit(this.$('input').val())
+        let value = parseFloat(this.get('value'))-1
+
+        if (!this.get('isAllowUseDownButtonToNegative')) {
```

Comment:
It might be interesting to disable the minus button instead of doing something like this?
We might simply just not decrease if the number is  === 0